server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -353,6 +350,35 @@ public void validateRepositoryCanBeCreated(final PutRepositoryRequest request) {
         closeRepository(createRepository(newRepositoryMetadata));
     }
 
+    private void validatePutRepositoryRequest(final PutRepositoryRequest request, ActionListener<Void> resultListener) {
+        final RepositoryMetadata newRepositoryMetadata = new RepositoryMetadata(request.name(), request.type(), request.settings());
+        try {
+            final var repository = createRepository(newRepositoryMetadata);
+            if (request.verify()) {
+                // verify repository on local node only, different from verifyRepository method that runs on other cluster nodes
+                threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(() -> {
+                    try {
+                        final var token = repository.startVerification();
+                        if (token != null) {
+                            repository.verify(token, clusterService.localNode());
+                            repository.endVerification(token);
+                        }
+                        resultListener.onResponse(null);
+                    } catch (Exception e) {
+                        resultListener.onFailure(e);
+                    } finally {
+                        closeRepository(repository);
+                    }
+                });
```

Comment:
It's generally best to avoid submitting a bare `Runnable` to threadpool executors, because they don't handle rejection very well. In this particular case it'd be ok since (a) `SNAPSHOT` doesn't reject anything (today) and (b) the whole thing is in a try/catch block anyway, but still these things are easy to miss in future changes in this area. `AbstractRunnable` is always a better choice:

```suggestion
                threadPool.executor(ThreadPool.Names.SNAPSHOT)
                    .execute(ActionRunnable.run(ActionListener.runBefore(resultListener, () -> closeRepository(repository)), () -> {
                        final var token = repository.startVerification();
                        if (token != null) {
                            repository.verify(token, clusterService.localNode());
                            repository.endVerification(token);
                        }
                    }));
```

Reply:
neat

Reply:
The try-catch block may not be necessary since we are calling this method in a `SubscribableListener#newForked` which does its own try-catch. That said, I am not too fussed with the additional try-catch. It helps reasoning when just reading this single method.